src/modules/GroupChannel/context/hooks/useMessageListScroll.tsx

```diff
@@ -25,41 +24,32 @@ export type ScrollTopicUnion =
       };
     };
 
-function runCallback(callback: () => void, lazy = true) {
-  if (lazy) {
-    setTimeout(() => {
-      callback();
-    });
-  } else {
-    callback();
-  }
-}
-
-function getScrollBehavior(behavior: 'smooth' | 'auto', animated?: boolean) {
-  if (typeof animated === 'boolean') return animated ? 'smooth' : 'auto';
-  return behavior;
-}
-
-export function useMessageListScroll(behavior: 'smooth' | 'auto') {
+export function useMessageListScroll(behavior: 'smooth' | 'auto', deps: DependencyList = []) {
   const scrollRef = useRef<HTMLDivElement>(null);
+  const scrollPositionRef = useRef(0);
   const scrollDistanceFromBottomRef = useRef(0);
 
-  const [scrollPubSub] = useState(() => pubSubFactory<ScrollTopics, ScrollTopicUnion>());
-  const [isScrollBottomReached, setIsScrollBottomReached] = useState(false);
+  const [scrollPubSub] = useState(() => pubSubFactory<ScrollTopics, ScrollTopicUnion>({ publishSynchronous: true }));
```

Comment:
Any reason we use useState here instead of useRef? It seems it doesn't have a set function.